Build scripts broken

I just tried compiling LFS in a clean Ubuntu 14.04 (trusty) Docker container and got an error. Something about package includes (sorry, I don't know much about GO). I got similar errors trying to build it on a Debian 8 machine. To reproduce use the following:

```
docker run -i -t --entrypoint /bin/bash ubuntu:trusty
apt-get install git
git clone https://github.com/github/git-lfs.git
cd git-lfs/
script/debian-build
```

which gives

```
<output omitted>
+ ./script/bootstrap
script/build.go:13:2: cannot find package "github.com/github/git-lfs/lfs" in any of:
	/root/.gvm/gos/go1.4.2/src/github.com/github/git-lfs/lfs (from $GOROOT)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/github/git-lfs/lfs (from $GOPATH)
++ echo FAIL
FAIL
```

On CentOS 7 following the same steps I get: 

```
script/build.go:13:2: cannot find package "github.com/github/git-lfs/lfs" in any of:
	/usr/lib/golang/src/github.com/github/git-lfs/lfs (from $GOROOT)
	($GOPATH not set)
FAIL
```

--- a/.gitignore
+++ b/.gitignore
@@ -16,3 +16,5 @@ obj-*
 
 rpm/BUILD*
 rpm/*RPMS
+
+src

--- a/script/bootstrap
+++ b/script/bootstrap
@@ -1,4 +1,5 @@
 #!/bin/sh
+set -e
 
 script/fmt
 rm -rf `pwd`/bin/*

--- a/script/bootstrap
+++ b/script/bootstrap
@@ -1,6 +1,13 @@
 #!/bin/sh
 set -e
 
+if [ -z "$GOPATH" ]; then
+  export GOPATH="$(pwd)"
+  mkdir -p src/github.com/github
+  [ -h src/github.com/github/git-lfs ] ||
+    ln -s "$GOPATH" src/github.com/github/git-lfs
+fi
+
 script/fmt
 rm -rf bin/*
 go run script/*.go -cmd build "$@"

--- a/script/bootstrap
+++ b/script/bootstrap
@@ -1,7 +1,9 @@
 #!/bin/sh
 set -e
 
-if [ -z "$GOPATH" ]; then
+# Only set GOPATH on non-Windows platforms as Windows can't do the symlinking
+# that we need.
+if [ -z "$GOPATH" ] && [ uname -s | grep -vq "_NT-" ]; then
   export GOPATH="$(pwd)"
   mkdir -p src/github.com/github
   [ -h src/github.com/github/git-lfs ] ||

--- a/script/bootstrap
+++ b/script/bootstrap
@@ -2,5 +2,5 @@
 set -e
 
 script/fmt
-rm -rf `pwd`/bin/*
+rm -rf bin/*
 go run script/*.go -cmd build "$@"

